dataprofiler/labelers/character_level_cnn_model.py

```diff
@@ -759,27 +759,21 @@ def predict(self, data, batch_size=32, show_confidences=False,
             allocation_index += num_samples_in_batch
 
         # Convert predictions, confidences to lists from numpy
-        predictions = predictions.tolist()
+        r_predictions = []
+        r_confidences = None
         if show_confidences:
-            confidences = confidences.tolist()
-        else:
-            confidences = None
-
-        # Trim array size to number of samples
-        if len(predictions) > allocation_index:
-            del predictions[allocation_index:]
-            if show_confidences:
-                del confidences[allocation_index:]
-
-        # Trim array size to length of sentence
+            r_confidences = []
+                
+        # Append slices of predictions to return prediction & confidence matrices
         for index, sentence_length \
                 in enumerate(sentence_lengths[:allocation_index]):
-            del predictions[index][sentence_length:]
+            r_predictions.append(list(predictions[index][:sentence_length]))
```

Comment:
I feel liek this could be optimized further. I'm not saying we should go back to `tolist` but if I recall, `del` is actually faster than slicing bc slicing creates a duplicate of the data whereas del would be in place and is almost free. Maybe array size dependent. 

https://stackoverflow.com/questions/56419072/most-efficient-way-to-remove-last-element-of-list

Could also be different for last X elements of an array.

Reply:
In addition we are appending to an empty array which will be slower than a preallocated array.